packages/MJCore/src/generic/metadata.ts

```diff
@@ -196,22 +197,21 @@ export class Metadata {
     }
 
     /**
-     * Returns the Name of the specific primaryKeyValues for a given entityName. This is done by 
+     * Returns the Name of the specific KeyValuePairs for a given entityName. This is done by 
      * looking for the IsNameField within the EntityFields collection for a given entity. 
      * If no IsNameField is found, but a field called "Name" exists, that value is returned. Otherwise null returned 
      * @param entityName 
-     * @param primaryKeyValues
+     * @param KeyValuePairs
      * @returns the name of the record
      */
-    public async GetEntityRecordName(entityName: string, primaryKeyValues: PrimaryKeyValue[]): Promise<string> {
+    public async GetEntityRecordName(entityName: string, CompositeKey: CompositeKey): Promise<string> {
         // check each primary key value to make sure it's not null
-        for (let j = 0; j < primaryKeyValues.length; j++) {
-            if (!primaryKeyValues[j] || !primaryKeyValues[j].Value) {
-                throw new Error('GetEntityRecordName: primaryKeyValues cannot contain null values. FieldName: ' + primaryKeyValues[j]?.FieldName);
+        for (let j = 0; j < CompositeKey.KeyValuePairs.length; j++) {
```

Comment:
@JS-BC  you should move this into a function INSIDE the CompositeKey class to `Validate` a given instance of CK so you don't have it here, and just call `Validate` for the passed in CK. 

ALSO - really bad practice to have the variable name exactly the same as the Class Name. I wish TS would throw a compile error at you for this. call it `compositeKey` or `ck` or whatever, just not exact same casing as the class name